patch/cisco-add-support-for-ltc2979-chip.patch

```diff
@@ -0,0 +1,144 @@
+(lowest = +3.30 V, highest = +3.30 V)
+vout7: +0.00 V (crit min = +1.10 V, min = +1.14 V)
+(max = +1.46 V, crit max = +1.50 V)
+(lowest = +0.00 V, highest = +0.00 V)
+vout8: +0.00 V (crit min = +1.27 V, min = +1.32 V)
+(max = +1.68 V, crit max = +1.72 V)
+(lowest = +0.00 V, highest = +0.00 V)
+
+Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+---
+ drivers/hwmon/pmbus/ltc2978.c | 13 ++++++++++---
+ 1 file changed, 10 insertions(+), 3 deletions(-)
+
+diff --git a/drivers/hwmon/pmbus/ltc2978.c b/drivers/hwmon/pmbus/ltc2978.c
+index 94eea2ac6..d43f2d0c8 100644
+--- a/drivers/hwmon/pmbus/ltc2978.c
++++ b/drivers/hwmon/pmbus/ltc2978.c
+@@ -27,8 +27,8 @@
+ #include <linux/regulator/driver.h>
+ #include "pmbus.h"
+ 
+-enum chips { ltc2974, ltc2975, ltc2977, ltc2978, ltc2980, ltc3880, ltc3882,
+-	ltc3883, ltc3886, ltc3887, ltm2987, ltm4675, ltm4676 };
++enum chips { ltc2974, ltc2975, ltc2977, ltc2978, ltc2979, ltc2980, ltc3880,
++	ltc3882, ltc3883, ltc3886, ltc3887, ltm2987, ltm4675, ltm4676 };
+ 
+ /* Common for all chips */
+ #define LTC2978_MFR_VOUT_PEAK		0xdd
+@@ -36,6 +36,8 @@ enum chips { ltc2974, ltc2975, ltc2977, ltc2978, ltc2980, ltc3880, ltc3882,
+ #define LTC2978_MFR_TEMPERATURE_PEAK	0xdf
+ #define LTC2978_MFR_SPECIAL_ID		0xe7	/* Undocumented on LTC3882 */
+ #define LTC2978_MFR_COMMON		0xef
++#define LTC2979_ID_A			0x8060
++#define LTC2979_ID_B			0x8070
+ 
+ /* LTC2974, LTC2975, LCT2977, LTC2980, LTC2978, and LTM2987 */
+ #define LTC2978_MFR_VOUT_MIN		0xfb
+@@ -503,6 +505,7 @@ static const struct i2c_device_id ltc2978_id[] = {
+ 	{"ltc2975", ltc2975},
+ 	{"ltc2977", ltc2977},
+ 	{"ltc2978", ltc2978},
++	{"ltc2979", ltc2979},
+ 	{"ltc2980", ltc2980},
+ 	{"ltc3880", ltc3880},
+ 	{"ltc3882", ltc3882},
+@@ -569,6 +572,8 @@ static int ltc2978_get_id(struct i2c_client *client)
+ 		return ltc2977;
+ 	else if (chip_id == LTC2978_ID_REV1 || chip_id == LTC2978_ID_REV2)
+ 		return ltc2978;
++	else if (chip_id == LTC2979_ID_A || chip_id == LTC2979_ID_B)
++		return ltc2979;
+ 	else if (chip_id == LTC2980_ID_A || chip_id == LTC2980_ID_B)
+ 		return ltc2980;
+ 	else if (chip_id == LTC3880_ID)
+@@ -668,6 +673,7 @@ static int ltc2978_probe(struct i2c_client *client,
+ 		break;
+ 	case ltc2977:
+ 	case ltc2978:
++	case ltc2979:
+ 	case ltc2980:
+ 	case ltm2987:
+ 		info->read_word_data = ltc2978_read_word_data;
+@@ -761,6 +767,7 @@ static const struct of_device_id ltc2978_of_match[] = {
+ 	{ .compatible = "lltc,ltc2975" },
+ 	{ .compatible = "lltc,ltc2977" },
+ 	{ .compatible = "lltc,ltc2978" },
++	{ .compatible = "lltc,ltc2979" },
+ 	{ .compatible = "lltc,ltc2980" },
+ 	{ .compatible = "lltc,ltc3880" },
+ 	{ .compatible = "lltc,ltc3882" },
+@@ -788,5 +795,5 @@ static struct i2c_driver ltc2978_driver = {
+ module_i2c_driver(ltc2978_driver);
+ 
+ MODULE_AUTHOR("Guenter Roeck");
+-MODULE_DESCRIPTION("PMBus driver for LTC2978 and comppatible chips");
++MODULE_DESCRIPTION("PMBus driver for LTC2978 and compatible chips");
```

Comment:
i do not see this change in the upstream patch, please remove.

Reply:
Hi Guhon,

Upstream patch has corrected the spelling mistake in the above code. 
This is the line from  torvalds/linux@ee44faf 5.7-rc1
"MODULE_DESCRIPTION("PMBus driver for LTC2978 and compatible chips");"

Please let me know if you still want me to remove this change.

Reply:
As discussed today, please create two patches with `git format-patch`:

    $ git format-patch -1 ee44fafb38869e89ff1012134ea80d55e6e346fc
    $ git format-patch -1 89688e8d1ecda59edc4e0ceb430aedaac18ca071

Reply:
Patches created with with format-patch does not apply clean with 202012 branch kernel [ 4.19.152 ]

Please approve this patch for 202012 branch. 

Will remove this patch when migrating to 5.10 kernel

Reply:
> Patches created with with format-patch does not apply clean with 202012 branch kernel [ 4.19.152 ]

What other commits are missing to get it applied cleanly?